### PR TITLE
Add ImageSet custom resource.

### PR DESCRIPTION
**Pull request by r0mant**
Add `ImageSet` custom resource as a first step of implementing https://docs.google.com/document/d/1ms8zW37DWOvH4nWPbQZxYBVxlkqQ1NOZ1f1KfQdxynY/edit?usp=sharing.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# [[override]]
+#   name = "github.com/x/y"
+#   version = "2.4.0"
+#
+# [prune]
+#   non-go = false
+#   go-tests = true
+#   unused-packages = true
+
+[[override]]
```
Comment (a-palchikov) [inline Gopkg.toml]: nit: why is this an override?

File: pkg/apis/lens/v1beta1/register.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// SchemeGroupVersion is group version used to register these objects.
+var SchemeGroupVersion = schema.GroupVersion{
+	Group:   lens.GroupName,
+	Version: "v1beta1",
+}
+
+// Kind takes an unqualified kind and returns back a Group qualified GroupKind.
```
Comment (a-palchikov) [inline pkg/apis/lens/v1beta1/register.go]: ~~back~~

File: pkg/apis/lens/v1beta1/register.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+// Resource takes an unqualified resource and returns a Group qualified GroupResource.
+func Resource(resource string) schema.GroupResource {
+	return SchemeGroupVersion.WithResource(resource).GroupResource()
+}
+
+var (
+	// SchemeBuilder initializes a scheme builder.
+	SchemeBuilder = runtime.NewSchemeBuilder(addKnownTypes)
+	// AddToScheme is a global function that registers this API group & version to a scheme.
```
Comment (a-palchikov) [inline pkg/apis/lens/v1beta1/register.go]: registers in?..

File: pkg/apis/lens/v1beta1/types.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+	metav1.TypeMeta `json:",inline"`
+	// ObjectMeta is the resource metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec is the image set specification.
+	Spec ImageSetSpec `json:"spec"`
+	// Status contains image set runtime information.
+	Status ImageSetStatus `json:"status"`
+}
+
+//ImageSetSpec is the image set specification.
```
Comment (a-palchikov) [inline pkg/apis/lens/v1beta1/types.go]: `// ImageSetSpec contains the list of docker image references`

File: resources/crd.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+spec:
+  group: cluster.gravitational.io
+  version: v2
+  scope: Cluster
+  names:
+    kind: ImageSet
+    plural: imagesets
+    singular: imageset
+    shortNames:
+    - img
```
Comment (a-palchikov) [inline resources/crd.yaml]: `imgset`? `img` does not really fit the list of image references

File: resources/crd.yaml [hunk 6]
```diff
@@ -0,0 +1,33 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: imagesets.cluster.gravitational.io
```
Comment (knisbet) [inline resources/crd.yaml]: not sure if I'm missing something, shouldn't this be imagesets.lens.gravitational.io (and below)?

File: pkg/apis/lens/register.go [hunk 7]
```diff
@@ -0,0 +1,22 @@
+/*
```
[the remark this answers is not in the document]
Comment (r0mant) [reply pkg/apis/lens/register.go]: I don't know, I structured it same way Kubernetes' own APIs are structured, e.g. `kube-aggregator/pkg/apis/apiregistration/v1`, `kube-aggregator/pkg/apis/apiregistration/v1` and so on. I haven't seen the full group name being used as a package name.

File: Gopkg.toml (hunk 1 above)
Comment (r0mant) [reply Gopkg.toml]: It is not used directly yet.

**Comment by r0mant** [comment]
@knisbet Addressed your comments.

**Comment by r0mant** [comment]
@a-palchikov @knisbet FYI talking to Sasha, it seems like it makes more sense to keep this in the Gravity repo at this point so I'm gonna close this PR and merge it into my Gravity one.

**Pull request closed without merging**